src/main/java/com/todoist/markup/MarkupParser.java

```diff
@@ -28,31 +19,31 @@
         List<MarkupEntry> markupEntries = new ArrayList<MarkupEntry>();
 
         if (string != null) {
-            if ((flags & HEADER) == HEADER) {
+            if ((flags & MarkupFlags.HEADER) == MarkupFlags.HEADER) {
                 parseHeaderMarkupEntries(string, markupEntries);
             }
 
-            if ((flags & BOLD) == BOLD) {
+            if ((flags & MarkupFlags.BOLD) == MarkupFlags.BOLD) {
                 parseBoldMarkupEntries(string, markupEntries);
             }
 
-            if ((flags & ITALIC) == ITALIC) {
+            if ((flags & MarkupFlags.ITALIC) == MarkupFlags.ITALIC) {
                 parseItalicMarkupEntries(string, markupEntries);
             }
 
-            if ((flags & LINK) == LINK) {
+            if ((flags & MarkupFlags.LINK) == MarkupFlags.LINK) {
                 parseLinkMarkupEntries(string, markupEntries);
             }
 
-            if ((flags & GMAIL) == GMAIL) {
+            if ((flags & MarkupFlags.GMAIL) == MarkupFlags.GMAIL) {
                 parseGmailMarkupEntries(string, markupEntries);
             }
 
-            if ((flags & OUTLOOK) == OUTLOOK) {
+            if ((flags & MarkupFlags.OUTLOOK) == MarkupFlags.OUTLOOK) {
                 parseOutlookMarkupEntries(string, markupEntries);
             }
 
-            if ((flags & THUNDERBIRD) == THUNDERBIRD) {
+            if ((flags & MarkupFlags.THUNDERBIRD) == MarkupFlags.THUNDERBIRD) {
                 parseThunderbirdMarkupEntries(string, markupEntries);
             }
```

Comment:
We're missing handling for the `EMOJI` flag. Something like:

```
if ((flags & MarkupFlags.EMOJI) == MarkupFlags.EMOJI) {
    parseEmojiMarkupEntries(string, markupEntries);
}
```

This would use `EmojiParser.getEmojiPattern()` to create markup entries for each of them. `MarkupApplier` would then perform the replacements of the original text with `MarkupEntry#text`, which contains the emoji itself.

This would effectively unify the API and let this library's clients use `MarkupParser` for emojis as well. I would even go as far as creating `MarkupParser#initAsync()` and `MarkupParser#getEmoji(String)` and proxying these calls to `EmojiParser`. This would hide `EmojiParser` from clients while still keeping it around for the future changes we plan to make.